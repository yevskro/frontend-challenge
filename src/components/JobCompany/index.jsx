import React from 'react';
import styled from 'styled-components';

function JobCompany({ children, newly, featured }) {
  return (
    <StyledJobCompany>
      {children}
      {newly ? <Newly /> : ''}
      {featured ? <Featured /> : ''}
    </StyledJobCompany>
  );
}

const Status = styled.span`
  line-height: 26px;
  vertical-align: middle;
  height: 24px;
  padding-left: 9px;
  padding-right: 9px;
  color: white;
  border-radius: 20px;
  margin-left: 4px;
  margin-right: 4px;
  font-size: ${({ theme }) => theme.fontSize.small};
`;

const Newly = styled(Status)`
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  margin-left: 18px;
  &:after {
    content: 'NEW!';
    font-size: ${({ theme }) => theme.fontSize.xxsmall};
  }
`;

const Featured = styled(Status)`
  background-color: ${({ theme }) => theme.color.neutral.veryDarkGrayishCyan};
  &:after {
    content: 'FEATURED';
    font-size: ${({ theme }) => theme.fontSize.xxsmall};
  }
`;

const StyledJobCompany = styled.div`
  display: flex;
  color: #5eaaa6;
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  height: 24px;
  align-items: center;
  font-size: ${({ theme }) => theme.fontSize.xsmall};
  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.small};
  }
`;
export default JobCompany;
