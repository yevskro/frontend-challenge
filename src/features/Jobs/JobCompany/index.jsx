import React from 'react';
import styled from 'styled-components';

function JobCompany({ children, newly, featured }) {
  return (
    <StyledJobCompany role="heading">
      {children}
      {newly ? <PrimaryPill role="note">NEW!</PrimaryPill> : ''}
      {featured ? <SecondaryPill role="note">FEATURED</SecondaryPill> : ''}
    </StyledJobCompany>
  );
}

const Pill = styled.span`
  line-height: 26px;
  vertical-align: middle;
  height: 24px;
  padding-left: 9px;
  padding-right: 9px;
  color: white;
  border-radius: 20px;
  margin-left: 4px;
  margin-right: 4px;
  font-size: ${({ theme }) => theme.fontSize.small};
`;

const PrimaryPill = styled(Pill)`
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  margin-left: 18px;
  font-size: ${({ theme }) => theme.fontSize.xxsmall};
`;

const SecondaryPill = styled(Pill)`
  background-color: ${({ theme }) => theme.color.neutral.veryDarkGrayishCyan};
  font-size: ${({ theme }) => theme.fontSize.xxsmall};
`;

const StyledJobCompany = styled.div`
  display: flex;
  color: #5eaaa6;
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  height: 24px;
  align-items: center;
  font-size: ${({ theme }) => theme.fontSize.xsmall};
  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.small};
  }
`;
export default JobCompany;
