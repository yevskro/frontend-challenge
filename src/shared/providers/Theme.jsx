import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
