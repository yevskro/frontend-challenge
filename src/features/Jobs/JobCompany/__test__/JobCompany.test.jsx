import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Theme from '../../../../shared/providers/Theme';
import JobCompany from '..';

test('renders', () => {
  render(
    <Theme>
      <JobCompany newly featured>
        Photosnap
      </JobCompany>
    </Theme>
  );

  expect(screen.getByText('Photosnap')).toBeInTheDocument();
  expect(screen.getByText('NEW!')).toBeInTheDocument();
  expect(screen.getByText('FEATURED')).toBeInTheDocument();
});

test('renders just newly pill', async () => {
  render(
    <Theme>
      <JobCompany newly />
    </Theme>
  );

  expect(screen.getByText('NEW!')).toBeInTheDocument();
  expect(screen.queryByText('FEATURED')).not.toBeInTheDocument();
});

test('renders featured pill', () => {
  render(
    <Theme>
      <JobCompany featured />
    </Theme>
  );

  expect(screen.queryByText('NEW!')).not.toBeInTheDocument();
  expect(screen.getByText('FEATURED')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <JobCompany newly featured>
          Durran
        </JobCompany>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
