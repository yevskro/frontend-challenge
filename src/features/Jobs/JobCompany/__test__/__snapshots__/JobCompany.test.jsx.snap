// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c1 {
  height: 24px;
  line-height: 26px;
  vertical-align: middle;
  padding: 0px 9px 0px 9px;
  margin: 0px 4px 0px 4px;
  color: white;
  border-radius: 20px;
  font-size: 0.9rem;
}

.c2 {
  margin-left: 18px;
  background-color: hsl(180,29%,50%);
  font-size: 0.7rem;
}

.c3 {
  background-color: hsl(180,14%,20%);
  font-size: 0.7rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  color: hsl(180,29%,50%);
  font-weight: 600;
  font-size: 0.8rem;
}

@media (min-width:500px) {
  .c0 {
    font-size: 0.9rem;
  }
}

<div
  className="c0"
>
  <h6>
    Durran
  </h6>
  <span
    className="c1 c2"
  >
    NEW!
  </span>
  <span
    className="c1 c3"
  >
    FEATURED
  </span>
</div>
`;
